frontend/src/scenes/data-warehouse/editor/editorSidebarLogic.tsx

```diff
@@ -62,139 +57,273 @@ export const editorSidebarLogic = kea<editorSidebarLogicType>([
             ['selectSourceTable', 'toggleJoinTableModal'],
         ],
     }),
+    loaders(({ values }) => ({
+        folders: [
+            [] as DataWarehouseFolder[],
+            {
+                loadFolders: async () => {
+                    const response = await api.dataWarehouseFolders.list()
+                    return response.results
+                },
+                addFolder: async ({ name, parentId }) => {
+                    const folder = await api.dataWarehouseFolders.create({ name, parent: parentId })
+                    return values.folders.concat(folder)
```

Comment:
I also attempted to add multiple new folders at the same time and ran into the duplicate name issue which I think could be resolved by some server side intelligence by returning something like `New folder (1)` like an operating system would.

Reply:
changed workflow to be require you to provide a name immediately like in a code editor